src/olympia/devhub/forms.py

```diff
@@ -586,10 +586,11 @@ def clean(self):
             self._clean_upload()
             xpi = parse_addon(self.cleaned_data['upload'], self.addon)
             # Make sure we don't already have the same non-rejected version.
-            if self.addon.versions.filter(version=xpi['version']).exclude(
-                    files__status=amo.STATUS_DISABLED):
+            if Version.unfiltered.filter(addon=self.addon,
+                                         version=xpi['version']):
```

Comment:
well, `.exists()` makes something along `select 1 from ...` and even if there's only one result it'll be much much faster and also will avoid any model-class initialization and more. The performance gain won't be big here I aggree but I rather simply use the tools we have (`.exists()`) than not. Especially if the change isn't that big and the gain can be noticable.